src/restaurant_inspections_api/routes.clj

```diff
@@ -1,27 +1,73 @@
-           (GET "/list-counties" [] (srv/get-dist-counties))
-           (GET ["/location/:zips" :zips #"[^/]+"]
-                [startDate endDate zips]
-                (srv/location startDate endDate zips))
-           (GET ["/name/:name"
-                 :name #"[^/]+"]
-                [startDate endDate name]
-                (srv/get-name startDate endDate name))
-           (GET ["/name/:name/:zips"
-                 :name #"[^/]+"
-                 :zips #"[^/]+"]
-                [startDate endDate name zips]
-                (srv/get-name startDate endDate name zips))
-           (GET "/district/:id"
-                [startDate endDate id]
-                (srv/district id startDate endDate))
-           (GET "/county/:id"
-                [startDate endDate id]
-                (srv/county id startDate endDate)))
 
+  (GET "/" [] (srv/home))
+
+  (ANY "/counties" []
+    (resource
+     :allowed-methods [:get]
+     :available-media-types ["application/json"]
+     ;; :handle-not-found	(fn [ctx] {:errors "No results found."})
+     ;; TODO: malformed when receiving query params
+     ;; :handle-malformed (400 BAD REQUEST)
+     :handle-ok (fn [ctx] (srv/get-counties))))
+
+  (ANY "/inspections/:id" [id] ;; id = inspections_visit_id
+    (resource
+     :allowed-methods [:get]
+     :available-media-types ["application/json"]
+     ;; :processable? inspections/inspections-processable?
+     ;; :handle-unprocessable-entity #(get % :errors-map)
+     ;; TODO: Not found when inspection id is not found (instead of 200)
+     ;; :handle-not-found	(fn [ctx] {:errors "No results found."})
+     ;; TODO: malformed when receiving query params
+     ;; :handle-malformed (400 BAD REQUEST)
+     :handle-ok (fn [ctx] (srv/full-inspection-details id))))
+
+  (ANY "/inspections" []
+    (resource
+     :allowed-methods [:get]
+     :available-media-types ["application/json"]
+     :processable? inspections/processable?
+     :handle-unprocessable-entity #(get % :errors-map)
+     ;; :handle-not-found (404 NOT FOUND)
+     ;; TODO: handle malformed when passing unknown query params
+     ;; :handle-malformed (400 BAD REQUEST)
+     ;; TODO: return count of returned data elements
+     :handle-ok inspections/handle-ok))
+
+  ;; TODO: Better api handling of businesses
+  (ANY "/businesses" []
+    (resource
+     :allowed-methods [:get]
+     :available-media-types ["application/json"]
+     ;; TODO: handle query params, the same way inspections do
+     ;; :processable? #(-> %)
+     ;;:handle-unprocessable-entity #(get % :errors-map)
+     :handle-ok (fn [ctx]
+                  (srv/get-businesses))))
+
+  ;; TODO: Better api handling of violation codes/definitions
+  (ANY "/violations" []
+    (resource
+     :allowed-methods [:get]
+     :available-media-types ["application/json"]
+     :processable? #(-> %)
+     :handle-unprocessable-entity #(get % :errors-map)
+     :handle-ok (fn [ctx]
+                  (srv/get-violations))))
```

Comment:
Add meta and data on 1st of level json. Right now inconsistent with inspections json structure.